sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -428,19 +428,36 @@ private void addTimedEvent(EventData timedEvent) {
   @Override
   public ReadWriteSpan setStatus(StatusCode statusCode, @Nullable String description) {
     if (statusCode == null) {
-      return this;
+      return this; // No action if statusCode is null
     }
     synchronized (lock) {
       if (!isModifiableByCurrentThread()) {
         logger.log(Level.FINE, "Calling setStatus() on an ended Span.");
-        return this;
-      } else if (this.status.getStatusCode() == StatusCode.OK) {
-        logger.log(Level.FINE, "Calling setStatus() on a Span that is already set to OK.");
-        return this;
+        return this; // Prevent modification if the span has ended
+      }
+
+      // Check the current status and enforce priority rules
+      StatusCode currentStatusCode = this.status.getStatusCode();
+
+      // Prevent setting a lower priority status.
+      if (currentStatusCode == StatusCode.OK) {
+        return this; // Do not allow lower priority status to override OK
+      } else if (currentStatusCode == StatusCode.ERROR && statusCode == StatusCode.UNSET) {
+        logger.log(Level.WARNING, "Cannot set status to UNSET when current status is ERROR.");
```

Comment:
I don't think this should be a WARNING. Probably just FINE like we have for the other status changes.